osis-core/src/main/java/com/scality/osis/platform/utils/ModelConverter.java

```diff
@@ -1,11 +1,66 @@
 package com.scality.osis.platform.utils;
 
+import com.scality.osis.model.OsisTenant;
+import com.scality.osis.model.exception.BadRequestException;
+import com.scality.vaultclient.dto.AccountData;
+import com.scality.vaultclient.dto.CreateAccountRequestDTO;
+import com.scality.vaultclient.dto.CreateAccountResponseDTO;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.scality.osis.platform.utils.ScalityConstants.*;
+
 /**
  * Convert between OSIS and Scality models
  */
 public final class ModelConverter {
 
+
     private ModelConverter() {
     }
 
+    public static CreateAccountRequestDTO toScalityAccountRequest(OsisTenant osisTenant) {
+        if(!osisTenant.getActive()){
+//            Scality does not support inactive tenant creation
+            throw new BadRequestException("Creating inactive tenant is not supported");
+        }
+        CreateAccountRequestDTO createAccountRequestDTO = new CreateAccountRequestDTO();
+
+        if (!StringUtils.isEmpty(osisTenant.getTenantId())) {
+            createAccountRequestDTO.setExternalAccountId(osisTenant.getTenantId());
+        }
+        createAccountRequestDTO.setName(osisTenant.getName());
+
+        // account email address == [cdTenantIds]@osis.account.com ; max cdTenantIds accepted is 6
+        // *** example email address 9b7e3259-aace-414c-bfd8-94daa0efefaf_9b7e3259-aace-414c-bfd8-94daa0efefaf@osis.account.com"
+        createAccountRequestDTO.setEmailAddress(toScalityAccountEmail(osisTenant.getCdTenantIds()));
+
+        return createAccountRequestDTO;
+    }
+
+    public static OsisTenant toOsisTenant(CreateAccountResponseDTO accountResponse) {
+        AccountData account = accountResponse.getAccount().getData();
+        return new OsisTenant()
+                .name(account.getName())
+                .active(true)
+                .cdTenantIds(toOsisCDTenantIds(account.getEmailAddress()))
+                .tenantId(account.getId());
+    }
+
+    public static List<String> toOsisCDTenantIds(String email) {
+        String ids = email.substring(0, email.indexOf(TENANT_EMAIL_SUFFIX));
+        return Arrays.asList(ids.split(ARRAY_SEPARATOR));
+    }
+
+    public static String toScalityAccountEmail(List<String> cdTenantIds) {
+//        extracting only first 6 ids
+        List<String> extractedCdTenantIds = cdTenantIds.stream().limit(MAX_ALLOWED_CD_TENANTID_COUNT).collect(Collectors.toList());
+        //TODO add condition for character count
+
+        return String.join(ARRAY_SEPARATOR, extractedCdTenantIds)
+                + "@osis.account.com";
```

Comment:
An email address in vault can max 256 characters.
Considering `MAX_ALLOWED_CD_TENANTID_COUNT` is 6, this will make the email address more than 256 characters.
Just curious, what was the reason to choose cd_tenant_id in email address?

Reply:
This is the initial design as we did not have any list or other field with maximum characters in the account. Only `emailAddress` seemed eligible to store all cd tenant ids. The `MAX_ALLOWED_CD_TENANTID_COUNT` as 6 is because of max allowed length for email address.

This will be changed in another ticket [S3C-4094](https://scality.atlassian.net/browse/S3C-4094)